### ISSUE TITLE
Cannot login to castopod 1.1.2 on yunohost anymore

### ISSUE THREAD

**Issue by simondueckert** [YunoHost-Apps/castopod_ynh#111]
### Describe the bug

*I have a yunohost instance and a castopod 1.1.2 installation at https://podcasts.cogneon.io. Since a few days I am not able to log into castopod anymore (with the same users that worked before). When I try to log in I get the error message "You could not be logged in. Please check your login details." (translated from german to english)*

### Context

- Hardware: *Virtual server at my provider (Hetzner)*
- YunoHost version: 1.1.2
- I have access to my server: *Through SSH | through the webadmin*
- Are you in a special context or did you perform some particular tweaking on your YunoHost instance?: *no*

### Steps to reproduce

- Open https://podcasts.cogneon.io/cp-admin
- Login with credentials

### Expected behavior

*Login should work like it did before. Where can I start to debug the problem? What might the problem be?*

### Logs

*I see nothing castopod specific in the logs.*

**Comment by orhtej2** [comment]
You didn't upgrade recently? It's possible to change the password via database or trick your system to allow magic link login if all else fails...

**Comment by simondueckert** [comment]
No, and I did not change the password. Can it be, that castopod lost the connection to the database? Where do I see the database credentials (I do not use docker)?

I cannot use the magick link since my castopod instance is not able to send emails.

**Comment by orhtej2** [comment]
You can retrieve credentials by issuing `sudo yunohost app setting castopod db_pwd`, user should be `castopod`.

You can also try adding `CI_ENVIRONMENT="development" ` to `/var/www/castopod/.env` to see more verbose error logs.

**Comment by orhtej2** [comment]
As for magic links edit e-mail config to use i.e. Your user in `.env` file

**Comment by TamerTemel** [comment]
Hi all,
I can see this is an old post but I have a workaround solution for e-mails.

## Discovery 
I have recently discovered Castopod and I'm currently experimenting with it on a self hosted VirtualMachine... 

### Castopod version 
To be specific, I am using Castopod v.1.13.0 as downloaded directly from offical web pages.

### Post installation
Following installation, I could not get the e-mail functions to work and this was a bit of a show stopper as you can't create new-users without e-mail...
The neaest solution I could find was to use a different account.. Or something to that effect. That was not very helpful and the thread was automatically closed after 15 days...

### The error 
I am not very familiar with the structure of Castopod but I managed to use the logs to ascertain what was stopping e-mails from working and discovered that there was a degree of replication in the mail-send proocess resulting in failures...

**ERROR - 2024-10-21 15:14:09 --> Email: sendWithSmtp throwed ErrorException: stream_socket_enable_crypto(): SSL/TLS already set-up for this stream in /home/user/podcast.domain-name.tld/vendor/codeigniter4/framework/system/Email/Email.php:1922**

Note that your user-name, domain-name and or location of this file will depend on where installedd and your credentials on the server, I.e. different to that above example. 

And the number 1922 at the end refers to line-number in Email.php file. That's where you need to navigate to, to make changes...

### Note
I will describe how I approached and steps I took to workarounnd this probelem below. 

Admittedly, this is a bit of a hack than an elegant solution but I disabled a section of the code in Email.php, that is the part reported as replica leading to errors. After disacbling the function, I was able to create new usres and the system did send me a Magic-Link, which then lead to creating a passord and being able to log-in.

# Prerequisites to fix this issue
You will need one of the following two ways to access the hosting server
- SSH (I use PuTTy, **you may need sudo credentials to do this**)
-- alternatively you can use WinSCP and navigate to folder containing Email.php - double click to open in Windows editor)
- File Manager [optional[] (alternate way to find the files that need modification - if your hosting provider allows this)
- choose favourite editor (VI is popular, I use nano)
- log-in to your Castopod as **admin** (in your browser, this is default account from **www.yourisntalldomain.tld/cp-admin** page following installation)

# Overview of action you need to take
- modify your .ENV file
- modify the Email.php file 

## Modifying your .ENV file
- The .ENV file needs to be modified so that you comment out anything beyond;
-- yor email-address (joeBloggs@hotmail.com)
-- from e-mail address (joeBloggs@hotmail.com)
-- subject (Castopod notification)
--mail-server  (jmail.hotmail.com)
comment out the other entries such as 'tls',

## Modify Email.php
- SSH into your server
- at command prompt, type nano -l /home/user/podcast.domain-name.tld/vendor/codeigniter4/framework/system/Email/Email.php
- find line 1922
- add a block comment using /* and */ around the following code, as shown;

  
```
/*  
      if ($this->SMTPCrypto === 'tls') {
            $this->sendCommand('hello');
            $this->sendCommand('starttls');
            $crypto = stream_socket_enable_crypto(
                $this->SMTPConnect,
                true,
                STREAM_CRYPTO_METHOD_TLSv1_0_CLIENT
                | STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT
                | STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT
                | STREAM_CRYPTO_METHOD_TLSv1_3_CLIENT
            );
            if ($crypto !== true) {
                $this->setErrorMessage(lang('Email.SMTPError', [$this->getSMTPData()]));
                return false;
            }

        }
*/
```

The objective is to address the error message **ErrorException: stream_socket_enable_crypto(): SSL/TLS already set-up for this stream** seen in the log file...Specifically **"already set-up"** part.

That is all. There is no need to reboot or restart anything. Simply head over to create-user page in your browser and you should be able to create new user(s). 

I have thrown together an [article about this on my blog pages](https://blog.tamertemel.net/2024/11/22/castopod-creating-a-new-user/#email-php-file) with images of screen-captures if you need a more detailed explanation.

I hope this is useful to anyone experiencing the same issue.

Reegards,
Tam..
